omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_0/AllergenConverter2_0.java

```diff
@@ -0,0 +1,100 @@
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_11.ConceptResource1_11;
+import org.apache.commons.lang3.StringUtils;
+import java.util.LinkedHashMap;
+
+/**
+ * An implementation of Converter to be able to create a representation from a User
+ * when User is used in another resource. Currently UserAndPassword doesn't convert User
+ */
+@Handler(supports = Allergen.class, order = 0)
+public class AllergenConverter2_0 implements Converter<Allergen> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Allergen getByUniqueId(String string) {
+		//return Context.getUserService().getUserByUuid(string);
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#newInstance(java.lang.String)
+	 */
+	@Override
+	public Allergen newInstance(String type) {
+		return new Allergen();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#asRepresentation(T, org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public SimpleObject asRepresentation(Allergen instance, Representation rep) throws ConversionException {
+		SimpleObject allergenObject = new SimpleObject();
+		allergenObject.add("allergenType", instance.getAllergenType());
+		ConceptResource1_11 conceptResource = (ConceptResource1_11) Context.getService(RestService.class).getResourceBySupportedClass(Concept.class);
+		allergenObject.add("codedAllergen", conceptResource.asRepresentation(instance.getCodedAllergen(), rep));
+		allergenObject.add("nonCodedAllergen", instance.getNonCodedAllergen());
+		return allergenObject;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#getProperty(T, java.lang.String)
+	 */
+	@Override
+	public Object getProperty(Allergen instance, String propertyName) throws ConversionException {
+		if (propertyName.equals("allergenType")) {
+			return instance.getAllergenType().toString();
+		} else if (propertyName.equals("codedAllergen")) {
+			return instance.getCodedAllergen();
+		} else {
+			return instance.getNonCodedAllergen();
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#setProperty(java.lang.Object, java.lang.String, java.lang.Object)
+	 */
+	@Override
+	public void setProperty(Object instance, String propertyName, Object value) throws ConversionException {
+		Allergen allergen = (Allergen)instance;
+		if (propertyName.equals("allergenType")) {
+			String aType = (String)value;
+			allergen.setAllergenType(StringUtils.isBlank(aType) ? null : AllergenType.valueOf(aType));
+		} else if (propertyName.equals("codedAllergen")) {
+			LinkedHashMap so = (LinkedHashMap)value;
+			Concept concept = Context.getConceptService().getConceptByUuid((String)so.get("uuid"));
+			allergen.setCodedAllergen(concept);
+		} else {
```

Comment:
Same here, you need to explicitly check the name for the nonCodedAllergen property